### PR TITLE
Issue #5759 Update jakarta transaction mail and inject apis

### LINKED ISSUE #5759: Update jakarta transaction, mail and injection apis
**jetty-11*

### PR DESCRIPTION
**Pull request by janbartel**
Closes #5759 

Update to final versions of RC releases for jakarta.transaction-api, jakarta.mail-api and jakarta.inject-api.

### CODE REVIEW

File: demos/demo-jndi-webapp/src/main/config/modules/demo-jndi.mod [hunk 1]
```diff
@@ -17,5 +17,4 @@ demo-mock-resources
 [files]
 basehome:modules/demo.d/demo-jndi.xml|webapps/demo-jndi.xml
 maven://org.eclipse.jetty.demos/demo-jndi-webapp/${jetty.version}/war|webapps/demo-jndi.war
-maven://jakarta.mail/jakarta.mail-api/2.0.0-RC4/jar|lib/ext/jakarta.mail-api-2.0.0-RC4.jar
-maven://jakarta.transaction/jakarta.transaction-api/2.0.0-RC1/jar|lib/ext/jakarta.transaction-api-2.0.0-RC1.jar
+maven://jakarta.mail/jakarta.mail-api/2.0.0/jar|lib/ext/jakarta.mail-api-2.0.0.jar
```
Comment (joakime) [inline demos/demo-jndi-webapp/src/main/config/modules/demo-jndi.mod:20]: Where did `jakarta.transaction` go?

**Review by joakime — COMMENTED** [review]
Is `jakarta.transaction` no longer needed here?

File: demos/demo-jndi-webapp/src/main/config/modules/demo-jndi.mod (hunk 1 above)
Comment (joakime) [reply demos/demo-jndi-webapp/src/main/config/modules/demo-jndi.mod:20]: Ah. I understand now.
Commit 3ae3e1fa47a3a226497e2be71c2b6ef56b851f49 helped.

**Review by joakime — APPROVED** [review]
LGTM

File: demos/demo-jndi-webapp/src/main/config/modules/demo-jndi.mod (hunk 1 above)
Comment (janbartel) [reply demos/demo-jndi-webapp/src/main/config/modules/demo-jndi.mod:20]: It should never have been put there, as that jar is in $jetty-home/lib and is put onto the classpath by `.plus.mod`

**Pull request merged**